<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="0" total="6" passed="1" failed="5" skipped="0">
  <reporter-output>
  </reporter-output>
  <suite started-at="2024-12-23T11:45:29 IST" name="Surefire suite" finished-at="2024-12-23T11:45:32 IST" duration-ms="2950">
    <groups>
      <group name="cucumber">
        <method signature="AbstractTestNGCucumberTests.runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:com.APIAutomation.runners.TestRunner@7722c3c3]" name="runScenario" class="io.cucumber.testng.AbstractTestNGCucumberTests"/>
      </group> <!-- cucumber -->
    </groups>
    <test started-at="2024-12-23T11:45:29 IST" name="Surefire test" finished-at="2024-12-23T11:45:32 IST" duration-ms="2950">
      <class name="com.APIAutomation.runners.TestRunner">
        <test-method is-config="true" signature="setUpClass(org.testng.ITestContext)[pri:0, instance:com.APIAutomation.runners.TestRunner@7722c3c3]" started-at="2024-12-23T11:45:29 IST" name="setUpClass" finished-at="2024-12-23T11:45:29 IST" duration-ms="504" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[org.testng.TestRunner@14f5da2c]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUpClass -->
        <test-method signature="runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:com.APIAutomation.runners.TestRunner@7722c3c3]" started-at="2024-12-23T11:45:29 IST" name="runScenario" description="Runs Cucumber Scenarios" data-provider="scenarios" finished-at="2024-12-23T11:45:32 IST" duration-ms="2287" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA["Book Creation by Different User Types"]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA["Book Creation Scenarios"]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- runScenario -->
        <test-method signature="runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:com.APIAutomation.runners.TestRunner@7722c3c3]" started-at="2024-12-23T11:45:29 IST" name="runScenario" description="Runs Cucumber Scenarios" data-provider="scenarios" finished-at="2024-12-23T11:45:32 IST" duration-ms="2309" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA["Book Creation by Different User Types"]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA["Book Creation Scenarios"]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Unexpected status code. Response body: Book Already Exists expected [201] but found [208]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Unexpected status code. Response body: Book Already Exists expected [201] but found [208]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertEqualsImpl(Assert.java:150)
at org.testng.Assert.assertEquals(Assert.java:132)
at org.testng.Assert.assertEquals(Assert.java:1419)
at org.testng.Assert.assertEquals(Assert.java:1383)
at com.APIAutomation.stepdefinitions.post.PostBooksStepDefinitions.verifyStatusCode(PostBooksStepDefinitions.java:63)
at ✽.the response status code should be 201(file:///D:/IT%20Quality%20Assuarance/Assignment/ITQA-API-Testing/src/test/resources/features/create_books.feature:9)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- runScenario -->
        <test-method signature="runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:com.APIAutomation.runners.TestRunner@7722c3c3]" started-at="2024-12-23T11:45:29 IST" name="runScenario" description="Runs Cucumber Scenarios" data-provider="scenarios" finished-at="2024-12-23T11:45:32 IST" duration-ms="2311" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA["Book Creation by Different User Types"]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA["Book Creation Scenarios"]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Unexpected status code. Response body: Book Already Exists expected [201] but found [208]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Unexpected status code. Response body: Book Already Exists expected [201] but found [208]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertEqualsImpl(Assert.java:150)
at org.testng.Assert.assertEquals(Assert.java:132)
at org.testng.Assert.assertEquals(Assert.java:1419)
at org.testng.Assert.assertEquals(Assert.java:1383)
at com.APIAutomation.stepdefinitions.post.PostBooksStepDefinitions.verifyStatusCode(PostBooksStepDefinitions.java:63)
at ✽.the response status code should be 201(file:///D:/IT%20Quality%20Assuarance/Assignment/ITQA-API-Testing/src/test/resources/features/create_books.feature:9)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- runScenario -->
        <test-method signature="runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:com.APIAutomation.runners.TestRunner@7722c3c3]" started-at="2024-12-23T11:45:29 IST" name="runScenario" description="Runs Cucumber Scenarios" data-provider="scenarios" finished-at="2024-12-23T11:45:32 IST" duration-ms="2285" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA["Delete a book by ID as admin"]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA["Delete a book"]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Delete request status code does not match expected:<200> but was:<403>]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Delete request status code does not match expected:<200> but was:<403>
at org.testng.AssertJUnit.fail(AssertJUnit.java:65)
at org.testng.AssertJUnit.failNotEquals(AssertJUnit.java:467)
at org.testng.AssertJUnit.assertEquals(AssertJUnit.java:88)
at org.testng.AssertJUnit.assertEquals(AssertJUnit.java:318)
at com.APIAutomation.stepdefinitions.delete.BookDeletionStepDefinition.verify_status_code(BookDeletionStepDefinition.java:63)
at ✽.the response delete status code should be 200(file:///D:/IT%20Quality%20Assuarance/Assignment/ITQA-API-Testing/src/test/resources/features/delete_books.feature:7)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- runScenario -->
        <test-method signature="runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:com.APIAutomation.runners.TestRunner@7722c3c3]" started-at="2024-12-23T11:45:29 IST" name="runScenario" description="Runs Cucumber Scenarios" data-provider="scenarios" finished-at="2024-12-23T11:45:32 IST" duration-ms="2309" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA["Book Creation by Different User Types"]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA["Book Creation Scenarios"]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Unexpected status code. Response body: Book Already Exists expected [201] but found [208]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Unexpected status code. Response body: Book Already Exists expected [201] but found [208]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertEqualsImpl(Assert.java:150)
at org.testng.Assert.assertEquals(Assert.java:132)
at org.testng.Assert.assertEquals(Assert.java:1419)
at org.testng.Assert.assertEquals(Assert.java:1383)
at com.APIAutomation.stepdefinitions.post.PostBooksStepDefinitions.verifyStatusCode(PostBooksStepDefinitions.java:63)
at ✽.the response status code should be 201(file:///D:/IT%20Quality%20Assuarance/Assignment/ITQA-API-Testing/src/test/resources/features/create_books.feature:9)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- runScenario -->
        <test-method signature="runScenario(io.cucumber.testng.PickleWrapper,io.cucumber.testng.FeatureWrapper)[pri:0, instance:com.APIAutomation.runners.TestRunner@7722c3c3]" started-at="2024-12-23T11:45:29 IST" name="runScenario" description="Runs Cucumber Scenarios" data-provider="scenarios" finished-at="2024-12-23T11:45:32 IST" duration-ms="2285" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA["Delete a book by ID as user"]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA["Delete a book"]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Delete request status code does not match expected:<401> but was:<200>]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Delete request status code does not match expected:<401> but was:<200>
at org.testng.AssertJUnit.fail(AssertJUnit.java:65)
at org.testng.AssertJUnit.failNotEquals(AssertJUnit.java:467)
at org.testng.AssertJUnit.assertEquals(AssertJUnit.java:88)
at org.testng.AssertJUnit.assertEquals(AssertJUnit.java:318)
at com.APIAutomation.stepdefinitions.delete.BookDeletionStepDefinition.verify_status_code(BookDeletionStepDefinition.java:63)
at ✽.the response delete status code should be 401(file:///D:/IT%20Quality%20Assuarance/Assignment/ITQA-API-Testing/src/test/resources/features/delete_books.feature:13)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- runScenario -->
        <test-method is-config="true" signature="tearDownClass()[pri:0, instance:com.APIAutomation.runners.TestRunner@7722c3c3]" started-at="2024-12-23T11:45:32 IST" name="tearDownClass" finished-at="2024-12-23T11:45:32 IST" duration-ms="100" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDownClass -->
      </class> <!-- com.APIAutomation.runners.TestRunner -->
    </test> <!-- Surefire test -->
  </suite> <!-- Surefire suite -->
</testng-results>
